src/com/b44t/messenger/rpc/Rpc.java

```diff
@@ -0,0 +1,120 @@
+package com.b44t.messenger.rpc;
+
+import com.b44t.messenger.DcJsonrpcInstance;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+
+public class Rpc {
+    private Map<Integer, CompletableFuture<JsonElement>> requestFutures = new ConcurrentHashMap<>();
+    private DcJsonrpcInstance dcJsonrpcInstance;
+    private int requestId = 0;
+    private Gson gson = new GsonBuilder().serializeNulls().create();
+
+    public Rpc(DcJsonrpcInstance dcJsonrpcInstance) {
+        this.dcJsonrpcInstance = dcJsonrpcInstance;
+    }
+
+    private void processResponse() throws JsonSyntaxException {
+        String jsonResponse = dcJsonrpcInstance.getNextResponse();
+        Response response = gson.fromJson(jsonResponse, Response.class);
+
+        if (response.id == 0) { // Got JSON-RPC notification, ignore
```

Comment:
For my understanding, what's a JSON-RPC notification?

Reply:
some legacy thing I think, in the past the RPC used to send unsolicited notifications with events so "notifications" here means incoming events that come without the client requesting them to the RPC server, we now have a new api to manually poll for events

Reply:
Would be nice to either replace `notification` with `event` or remove this block.

Reply:
done